osu.Game.Tests/Visual/Gameplay/TestScenePlayerLoader.cs

```diff
@@ -97,6 +118,61 @@ public void TestModReinstantiation()
             AddAssert("player mods applied", () => playerMod2.Applied);
         }
 
+        [Test]
+        public void TestMutedNotification()
+        {
+            AddStep("set master volume to 0%", () => audioManager.Volume.Value = 0);
+            AddStep("reset notification lock", () => PlayerLoader.ResetNotificationLock());
+            //AddStep("reset notification overlay", () => notificationOverlay);
+            AddStep("load player", () => ResetPlayer(false));
+            AddUntilStep("wait for player", () => player.IsLoaded);
+
+            AddAssert("check for notification", () => container.NotificationOverlay.UnreadCount.Value == 1);
+            AddAssert("click notification", () =>
+            {
+                var scrollContainer = container.NotificationOverlay.Children.Last() as OsuScrollContainer;
+                var flowContainer = scrollContainer.Children.First() as FillFlowContainer<NotificationSection>;
+                return flowContainer.Children.First().First().Click();
+            });
+            AddAssert("check master volume", () => audioManager.Volume.IsDefault);
+
+            AddStep("restart player", () =>
+            {
+                var lastPlayer = player;
+                player = null;
+                lastPlayer.Restart();
+            });
+        }
+
+        private class TestPlayerLoaderContainer : Container
+        {
+            private TestPlayerLoader loader;
```

Comment:
useless.

Reply:
This is one approach peppy told me I could do, and the only one which seems to work with resetting.